# Heroes Of Pymoli
### Introductory Data Analysis project using Python Programming

Welcome to the Heroes of Pymoli! This is a fantasy world (much like World of Warcraft) where users create avatars, fight monsters, level up, and, most importantly, gear up to fight around the world of Pymoli. However... Some players are a bit more serious about the game than others. 

Below is my data analysis of certains trends associated with the game. I answer questions like "Which gender is more likely to spend money on the game?" "How much are players willing to spend on a specific item?" & "What age is the majority of the games audience?"

Ideally, the skills shown in this project would be used for data wrangling, mining, and visualization for companies that need a direction in marketing campaigns.

In [25]:
import pandas as pd
import os, csv
import numpy as np

In [26]:
csvpath = os.path.join('Resources/purchases.csv')
dataframe = pd.read_csv(csvpath)
dataframe.head(10)

Purchase ID             SN  Age                 Gender  Item ID  \
0            0        Lisim78   20                   Male      108   
1            1    Lisovynya38   40                   Male      143   
2            2     Ithergue48   24                   Male       92   
3            3  Chamassasya86   24                   Male      100   
4            4      Iskosia90   23                   Male      131   
5            5        Yalae81   22                   Male       81   
6            6      Itheria73   36                   Male      169   
7            7    Iskjaskst81   20                   Male      162   
8            8      Undjask33   22                   Male       21   
9            9    Chanosian48   35  Other / Non-Disclosed      136   

                                   Item Name  Price  
0  Extraction, Quickblade Of Trembling Hands   3.53  
1                          Frenzied Scimitar   1.56  
2                               Final Critic   4.88  
3                                Blindscythe   3.27  
4                                       Fury   1.44  
5                                  Dreamkiss   3.61  
6     Interrogator, Blood Blade of the Queen   2.18  
7                              Abyssal Shard   2.67  
8                                  Souleater   1.10  
9               Ghastly Adamantite Protector   3.58

## I. What is the total number of purchases in the dataset?

In [27]:
total_purchases = len(dataframe["Purchase ID"].value_counts())
total_purchases_df = pd.DataFrame([total_purchases], columns = ["TOTAL PURCHASES"])
total_purchases_df

TOTAL PURCHASES
0              780

This data set spans 780 purchases. The total amount of purchases should be displayed in the analysis as a base.

## II. How many players are there in our current data set?

In [28]:
players = len(dataframe["SN"].value_counts())
totalplayers = pd.DataFrame([players], columns = ["TOTAL PLAYERS"])
totalplayers.head()

TOTAL PLAYERS
0            576

Although the CSV we uploaded has 770+ transactions, the total amount of players that purchased item's is only 576. This is because some players were repeat customers. Because Pymoli experiences repeat customers, we need to count the unique values of "SN", the player's unique username.

## III. Out of all the items within the game, how many did players actually purchase?

In [31]:
purchaseditems = len(dataframe["Item Name"].value_counts())
purchaseditems_df = pd.DataFrame([purchaseditems], columns = ["Unique Items Purchased"])
purchaseditems_df

Unique Items Purchased
0                     179

There are a total of 179 unique items that were purchased by players. Here we use the same concept in question I. There are numerous items in the game, but other items are purchased more frequently than others. Therefore, we would need to count the number of unique items in the entire dataset. 

## IV. What is the average amount spent on a single item?

In other words, what is the mean amount that the creators of Heroes of Pymoli make off of a single purchase?

In [33]:
average = dataframe["Price"].mean()
averageprice = round(average, 2)
averageprice_df = pd.DataFrame([averageprice], columns = ["Average Price"])
averageprice_df

Average Price
0           3.05

## V. What was the Total Revenue indicated in this dataset?

In [34]:
revenue = dataframe["Price"].sum()
totalrevenue = pd.DataFrame([revenue], columns = ["Total Revenue"])
totalrevenue

Total Revenue
0        2379.77

Total Revenue is calculated by adding up all the prices spent for each purchase. A simple line of code that shows the total amount gained based off of purchases. Revenue is $2,379.77

## Display the Overall Analysis of all questions answered from I - V

In [41]:
analysis = pd.DataFrame({"TOTAL PLAYERS": [players],
                        "UNIQUE ITEMS PURCHASED": [purchaseditems],
                        "AVERAGE PRICE": [averageprice],
                        "TOTAL PURCHASES": [total_purchases],
                        "TOTAL REVENUE": [revenue]})
analysis.head()

TOTAL PLAYERS  UNIQUE ITEMS PURCHASED  AVERAGE PRICE  TOTAL PURCHASES  \
0            576                     179           3.05              780   

   TOTAL REVENUE  
0        2379.77